### DISCUSSION TITLE
Writing original image path and name in DOCX files

### DISCUSSION THREAD

**Discussion by ptram** [jgm/pandoc#9900] [Q&A] [answered]
Hi,

When converting a Markdown document into something like a DOCX one, I would like the original file names and file paths to be preserved. This would be useful to relink to alternative versions of the images, or to update them. It would also be useful when importing the DOCX file into a page layout program, where the links can be then rebuilt with a simple command.

In other words, I would like that something like this:

`![](images/my-beautiful-paysage.jpg)`

remains 'my-beautiful-paysage.jpg' in Word, with the link pointing to an 'image' folder in the same directory as the generated DOCX file.

Is this possible?

**Comment by massifrg** [comment]
Try this filter (I have not tested it):

```lua
return {
  {
    Image = function(image)
      local filename = pandoc.path.filename(image.src)
      local link_content = pandoc.Str(filename)
      local link = pandoc.Link(link_content, image.src, filename)
      return link
      -- if you want both the Image and the Link use this:
      -- return pandoc.List({image, link})
    end
  }
}
```

save it in a file `images_as_links.lua`, and the use it like this:

```sh
pandoc -f markdown -t docx -o test.docx -L images_as_links.lua test.md
```

**Reply by ptram** [reply]
Thank you very much! But I see I was not clear at all when describing my needs.

The beautiful filter inserts the image link as text in the Word document. And, optionally, it inserts an rtf image with the name rewritten in the internal Word rules ("rld[#]").

What I meant was instead an image linked, not embedded, with the link stored together with the image. This is the type of images that you insert, in Word, using the `Insert > Pictures > Picture from file` command, with the `Link to file` option selected.

In my tests, this type of image preserves its original name in the Word document, and when the Word document is imported into InDesign or Affinity Publisher. This means that the original images (or their exact copy) are still the ones linked to the document.

By examining the DOCX document, the "documents.xml.rels" contains the complete (absolute) original link to the image, something like:

`Target="file:////Users/username/Documents/Project/images/my-image.png" TargetMode="External"/></Relationships>
`

That's what I was happy to achieve with the conversion from Markdown to DOCX.

Thank you again!

**Comment by iandol** [comment, marked as answer]
Please see https://github.com/jgm/pandoc/issues/9815 — ODT files can easily preserve the images as links rather than embeds, in fact the opendocument writer that ODT is based on always links images. It should be possible to use opendocument to ODT and have properly linked images. #9815 shows the XML code is simple and so a filter could inject linked figures into ODT directly as a second option. ODT can be converted to DOCX if you must use DOCX.

As you mentioned, DOCX's OpenXML uses a much more complex markup for linked images that depends on a separate file, this is more challenging to solve with a filter for DOCX output directly. I imagine this requires direct support from the DOCX writer in Pandoc...
